src/misc.ts

```diff
@@ -57,8 +58,19 @@ export async function httpHeadHeader (url: string): Promise<http.IncomingHttpHea
     }
 
     return new Promise<http.IncomingMessage>((resolve, reject) => {
-      request(options, resolve)
-        .on('error', reject)
+      let res: http.IncomingMessage | null = null
+      const req = request(parsedUrl, options, (response) => {
+        res = response
+        resolve(res)
+      })
+        .once('error', reject)
+        .setTimeout(HTTP_TIMEOUT, () => {
+          const e = new Error('Http request timeout!')
+          if (res) {
+            res.destroy(e)
```

Comment:
Can we just emit the error at here, instead of destroy the `res`?

Reply:
No. If we just emit error, res and req won't close themselves, so the code will still get stuck, causing a memory leak.